File: src/main/kotlin/io/github/dylmeadows/eontimer/model/timer/Gen5TimerMode.kt
package io.github.dylmeadows.eontimer.model.timer

import io.github.dylmeadows.commonkt.javafx.util.Choice

enum class Gen5TimerMode(
    override val displayName: String
) : Choice {
    STANDARD("Standard"),
    C_GEAR("C-Gear"),
    ENTRALINK("Entralink"),
    ENHANCED_ENTRALINK("Entralink+");
}

File: src/models/Gen5TimerMode.cpp
//
// Created by Dylan Meadows on 2020-03-19.
//

#include "Gen5TimerMode.h"

namespace EonTimer {
    const std::vector<Gen5TimerMode> &getGen5TimerModes() {
        static const std::vector<Gen5TimerMode> values{STANDARD, C_GEAR, ENTRALINK, ENTRALINK_PLUS};
        return values;
    }

    const std::string &getName(const Gen5TimerMode mode) {
        static const std::string names[COUNT] = {"Standard", "C-Gear", "Entralink", "Entralink+"};
        return names[mode];
    }
}  // namespace EonTimer
